Lesson8/src/BuilderCatalog.java

```diff
@@ -0,0 +1,81 @@
+import java.io.File;
+import java.io.FileReader;
+import java.io.FileWriter;
+import java.io.IOException;
+import java.util.Arrays;
+import java.util.Objects;
+
+public class BuilderCatalog {
+    private String nameCatalog;
+    private File catalog;
+
+    public File getCatalog() {
+        return catalog;
+    }
+
+    public BuilderCatalog(String nameCatalog) {
+        this.nameCatalog = nameCatalog;
+        File catalog = new File(nameCatalog);
+        catalog.mkdir();
+        this.catalog = catalog;
+
+    }
+
+    public void renameCatalog(String newName) {
+        File newCatalog = new File(newName);
+        if (this.catalog.renameTo(newCatalog)) {
+            deletedCatalog(this.catalog);
+            this.nameCatalog = newName;
+            this.catalog = newCatalog;
+        } else {
+            System.out.println("It's not OK");
+        }
+    }
+
+    public static void deletedCatalog(File file) {
+        if (!file.exists())
+            return;
+        if (file.isDirectory()) {
+            for (File f : Objects.requireNonNull(file.listFiles())) {
```

Comment:
Objects.requireNonNull will throw an NPE if file.listFiles() is null) 
Is it expected behaviour?